src/main/java/com/endava/internship/collections/StudentSet.java

```diff
@@ -2,49 +2,136 @@
 
 import java.util.Collection;
 import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
 import java.util.Set;
 
 public class StudentSet implements Set<Student> {
+    private Node root;
+    private List<Student> list;
+    private int size = 0;
+
+    public StudentSet() {
+        list = new LinkedList<>();
+    }
+
     @Override
     public int size() {
-        //TODO
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        //TODO
-        return false;
+        return size == 0;
     }
 
     @Override
     public boolean contains(Object o) {
```

Comment:
Always mark method arguments as final if method is not used for changing them.
With final you guarantee that args will be save from changing
```suggestion
    public boolean contains(final Object o) {
```
Same for others